### PR TITLE
[rebased] Builder's communication with API finished and secured

### PR DESCRIPTION
**Pull request by BourgoisMickael**

### CODE REVIEW

File: builder/src/communication.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    this.endEndpoint = '/end'
+    this.stdoutBuffer = new Buffer()
+    this.stderrBuffer = new Buffer()
+  }
+
+  async _req (uri, body) {
+    const options = {
+      method: 'PUT',
+      uri,
+      json: (body && true) || false,
```
Comment (Arignir) [inline builder/src/communication.js]: Why `&& true`?

File: builder/src/config/config.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   nbuild_path: 'path_to_nbuild',
   build_api_url: 'URL_OF_THE_API/',
-  manifest_dir: '/path/to/manifest/directory'
+  manifest_dir: '/path/to/manifest/directory',
+  buffer_size: 2048
```
Comment (Arignir) [inline builder/src/config/config.js:5]: :+1:

File: server/builder/receiver.js [hunk 3]
```diff
@@ -1,19 +1,17 @@
-const Queue = require('./queue')
+const Queue = require('../rabbitmq')
```
Comment (Arignir) [inline server/builder/receiver.js]: Hmm I'm not sure to understand why this file is here. It calls `./nbuild.py` but is in the `server/` directory. Am I missing something?

File: server/builder/receiver.js [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
 
-// TODO: Bufferize the output to send 4096 char at a time to not send too much requests
-
 async function receiver () {
   const queue = new Queue('build-raven-os-org')
 
   await queue.receive(async (msg) => {
     const data = [...msg.content]
-    const url = config.build_api_url + 'manifest/'
+    const url = config.url + 'manifest/'
```
Comment (Arignir) [inline server/builder/receiver.js]: Don't you have removed this configuration from `server/src/config/config.js` in this commit?

**Review by Arignir — CHANGES_REQUESTED** [review]

File: builder/src/communication.js (hunk 1 above)
Comment (BourgoisMickael) [reply builder/src/communication.js]: the json property is used to say the body is in json format. However if there is no body (null) and the property json is set to true, the client (api) will try to parse the json body (which is null) and will throw an error. So before saying if the body is in json we check that there is a body

File: server/builder/receiver.js (hunk 3 above)
Comment (BourgoisMickael) [reply server/builder/receiver.js]: builder folder deleted. It was added by the rebase.

File: server/builder/receiver.js (hunk 4 above)
Comment (BourgoisMickael) [reply server/builder/receiver.js]: builder folder removed from server folder. It was forgotten after the rebase.

File: builder/src/communication.js (hunk 1 above)
Comment (Arignir) [reply builder/src/communication.js]: Why not using `body !== null`?

Comment (BourgoisMickael) [reply builder/src/communication.js]: Fixed.

**Review by Arignir — APPROVED** [review]

**Pull request merged**